compiler/injectdestructors.nim

```diff
@@ -791,6 +791,12 @@ proc p(n: PNode; c: var Con; mode: ProcessMode): PNode =
       if mode == normal and isRefConstr:
         result = ensureDestruction(result, c)
     of nkCallKinds:
+      let inSpawn = c.inSpawn
+      if n[0].kind == nkSym and n[0].sym.magic == mSpawn:
+        c.inSpawn.inc
+      elif c.inSpawn > 0:
```

Comment:
Suspicious logic. Why at the end of the ``of nkCallkinds`` handling code like

```nim

if n[0].kind == nkSym and n[0].sym.magic == mSpawn:
  c.inSpawn.dec
```

?

Reply:
The idea is that spawn can be written in the following way:
```nim
spawn myproc(call1(), (let x = call2(); call3(x, 0))
```
I want  `c.inSpawn` to be greater than zero only for `myproc` call but not for `call2` and `call3` in example above.